File: python/mario_less.py
# This program builds a one sided pyramid to the height of the user input

from cs50 import get_int


# Create function for getting height from user


def get_height(prompt):
    while True:
        n = get_int(prompt)
        if n > 0 and n < 9:
            return n


# Get height from user
height = get_height("Height: ")

# Build pyramid
for i in range(height):
    for j in range((height - 1) - i):
        print(" ", end="")
    for k in range(i + 1):
        print("#", end="")

File: pset1/mario_less.c
// This program asks the user to input the height and then prints a half pyramid to that height
#include <stdio.h>
#include <cs50.h>
 
int main(void)
{
    // Declaring variables
    int height, row, space, hash;
 
    // Asks user to input an integer 1 through 8
    // While the user input is incorrect it will continue to ask the user for input
    do
    {
        height = get_int("Enter height (1 through 8): ");
    }
    while (height < 1 || height > 8);
 
    // Print new row as long as the number of rows is less than the height
    for (row = 0; row < height; row++) 
    {   // Subtract one space from each row
        for (space = height - row; space > 1; space--)
        {   // Print a space
            printf(" ");
        } 
        // Add one hash to each row 
        for (hash = 1; hash <= row + 1; hash++)
        
        {   // Print a hash 
            printf("#");
        }
        // Print a new line
        printf("\n");    
    }
    return 0;
}